src/functions.php

```diff
@@ -231,6 +232,19 @@ function ($reason, $idx, Promise $aggregate) {
         ksort($results);
         return $results;
     });
+
+    if (true === $recursive) {
+        $promise = $promise->then(function ($results) use ($recursive, &$promises) {
+            foreach ($promises AS $promise) {
+                if (\GuzzleHttp\Promise\PromiseInterface::PENDING === $promise->getState()) {
+                    return all($promises, $recursive);
```

Comment:
Hi there, 

I checked again when using all(), new EachPromise() and combining them, and the onFulfill callbacks aren't called twice.
Try this:
```php
require_once __DIR__ . '/vendor/autoload.php';

use GuzzleHttp\Promise\EachPromise;
use GuzzleHttp\Promise\FulfilledPromise;
use GuzzleHttp\Promise\Promise;
use GuzzleHttp\Promise\PromiseInterface;
use function GuzzleHttp\Promise\all;
use GuzzleHttp\Promise\RejectedPromise;

// Create food promises using "new EachPromise()"
$config         = [
    'fulfilled' => function ($value) {
        printf('I love %s' . PHP_EOL, $value);
    },
    'rejected'  => function ($value) {
        printf('I hate %s' . PHP_EOL, $value);
    },
];
$foodPromises   = new ArrayIterator();
$foodPromises[] = new FulfilledPromise('vegetables');
$foodPromises[] = new Promise(function (PromiseInterface $promise) use (&$foodPromises) {

    $promise->resolve('hamburgers');

    // Add another promise into the stack during resolution
    $foodPromises[] = new Promise(function (PromiseInterface $promise) {
        $promise->resolve('burritos - but no one cares'); // should not be called
    });

});
$foodPromises[] = new RejectedPromise('cabbage');
$foodPromise    = (new EachPromise($foodPromises, $config))->promise();

// Now create drink promises using "all()" function
$drinkPromises   = new ArrayIterator();
$drinkPromises[] = new FulfilledPromise('beer');
$drinkPromises[] = new Promise(function (PromiseInterface $promise) use ($drinkPromises) {

    $promise->resolve('whisky');

    // Add another promise into the stack during resolution
    $drinkPromises[] = new Promise(function (PromiseInterface $promise) {
        $promise->resolve('diet coke - but no one cares'); // should be called
    });

});
$drinkPromise    = all($drinkPromises, true) // Using all() function with recursion
    ->then(function ($results) use ($config) {
        array_walk($results, $config['fulfilled']);
    })
    ->otherwise(function ($reason) use ($config) {
        call_user_func($config['rejected'], $reason);
    });

// Combine all promises and wait
all([$foodPromise, $drinkPromise])->wait();
```

It should output:
```
I love vegetables
I hate cabbage
I love hamburgers
I love beer
I love whisky
I love diet coke - but no one cares
```